src/XMakeTasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1530,6 +1530,10 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <ItemGroup>
       <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.Identity)' == '%(Identity)'">
         <SetTargetFramework>$(_ProjectReferenceTargetFrameworkProperties)</SetTargetFramework>
+
+        <UndefineProperties Condition="$(_ProjectReferenceTargetFrameworkProperties.Contains(`ProjectHasSingleTargetFramework=true`))">%(_MSBuildProjectReferenceExistent.UndefineProperties);TargetFramework;ProjectHasSingleTargetFramework</UndefineProperties>
```

Comment:
That kind of thing (returning a tuple) would have been ideal, but I don't see a way to do it. The MSBuild task only has one output parameter, and it can be assigned either to an Item or a Property.

Returning an item has the advantage of being structured data, but it runs somewhat afoul of the current nature of assigning these properties. I put that together at https://github.com/dotnet/sdk/pull/797#issuecomment-277128140 -- do you like that better? It's better in some ways (this "undefined, or unconditionally undefined" is definitely ugly), but worse in others (clearing the item so we can look at it one at a time, extracting the metadata for insertion into the referring project's metadata). I think on balance I like this better but I could be persuaded . . .